docs/factorial.md

```diff
@@ -0,0 +1,30 @@
+# Factorial
+
+The factorial of an integer `n` is defined as:
+
+`n! = 1 x 2 x 3 x 4 x ... x n`
+
+With the special case `0! = 1`.
+
+## Requirements
+
+You must write an executable program that accepts an integer `n` on standard input via the command line,
+and outputs `n!` to standard output.
+
+Note that the factorial function grows very quickly. For example, `4! = 24` but `8! = 40320`.
+Therefore, you should impose a limit on the input, so that the largest factorial still fits 
+into your language's largest supported datatype.
+
+Also note that the factorial is not defined for negative integers.
+
+## Testing
+
+Some tests for your program are:
+
+|   n   |   n!    |
+| :---: | :-----: |
```

Comment:
For the sake of completeness and consistency, can you add a column here with descriptions of your test cases. Can you also add the empty input case as well as the invalid input case? No input and invalid input can return the same error.

Reply:
I usually do something like:

| Description | Input | Output |
|--------------|-------|---------|
| Empty Input |         | "Usage: please supply an integer" |